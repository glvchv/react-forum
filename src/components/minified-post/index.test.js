import React from 'react';
import renderer from 'react-test-renderer';
import MinifiedPost from '.';
import AuthContext from '../../context/authContext';
import { BrowserRouter } from 'react-router-dom';

const post = {
    author: {
        username: 'vld',
        avatarUrl: 'http://localhost:0000/test'
    },
    _id: '123',
    likes: ['a', 'b', 'c'],
    category: 'testing',
    replies: ['a', 'b', 'c'],
    title: 'Test',
    date: '10.08.2020',
    text: "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book."
}

describe('Minified-post component', () => {
    it('should render component for guests', () => {
        const tree = renderer.create(
            <BrowserRouter>
                <MinifiedPost post={post} />
            </BrowserRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render component for auth users', () => {
        const tree = renderer.create(
            <AuthContext.Provider value={{
                user: {
                    loggedIn: true
                }
            }}>
                <BrowserRouter>
                    <MinifiedPost post={post} />
                </BrowserRouter>
            </AuthContext.Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
})