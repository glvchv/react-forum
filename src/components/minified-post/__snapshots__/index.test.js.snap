// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Minified-post component should render component for auth users 1`] = `
<div
  className="post-body"
>
  <div
    className="post-details"
  >
    <img
      alt="avatar"
      className="avatar"
      src="http://localhost:0000/test"
    />
    <p>
      vld
    </p>
    <small>
      8.10.2020 г., 0:00:00
    </small>
  </div>
  <div
    className="main"
  >
    <h2>
      <a
        className="title"
        href="/posts/123"
        onClick={[Function]}
      >
        Test
      </a>
    </h2>
    <p
      className="text"
    >
      Lorem Ipsum is simply dummy text of the printing and typesetting indus...
    </p>
    <hr />
    <div
      className="actions"
    >
      <p>
        3
         
        <small>
          likes
        </small>
      </p>
      <p>
        <small>
          testing
        </small>
      </p>
      <p>
        3
         
        <small>
          replies
        </small>
      </p>
    </div>
  </div>
</div>
`;

exports[`Minified-post component should render component for guests 1`] = `
<div
  className="post-body"
>
  <div
    className="post-details"
  >
    <img
      alt="avatar"
      className="avatar"
      src="/default.png"
    />
    <p>
      vld
    </p>
    <small>
      8.10.2020 г., 0:00:00
    </small>
  </div>
  <div
    className="main"
  >
    <h2
      className="title"
    >
      Test
    </h2>
    <p
      className="text"
    >
      Lorem Ipsum is simply dummy text of the printing and typesetting indus...
    </p>
    <hr />
    <div
      className="actions"
    >
      <p>
        3
         
        <small>
          likes
        </small>
      </p>
      <p>
        <small>
          testing
        </small>
      </p>
      <p>
        3
         
        <small>
          replies
        </small>
      </p>
    </div>
  </div>
</div>
`;
